src/classes/InstallScript.cls

```diff
@@ -28,9 +28,33 @@
     POSSIBILITY OF SUCH DAMAGE.
 */
 public without sharing class InstallScript implements InstallHandler {
+    private InstallContext context;
+    @TestVisible
+    private TelemetryService telemetryService {
+        get {
+            if (telemetryService == null) {
+                telemetryService = new TelemetryService();
+            }
+
+            return telemetryService;
+        }
+        set;
+    }
 
     public void onInstall(InstallContext context) {
+        this.context = context;
+
         insertPackageSettings();
+
+        if (isNewInstall()) {
+            return;
+        }
+
+        sendUsageMetrics();
```

Comment:
Do we need a try/catch here to ensure that potential failures don't affect the ability to upgrade?